### DISCUSSION TITLE
How set set Doctrine Defaults to Nullable (or disable NotNull)

### DISCUSSION THREAD

**Discussion by tacman** [symfony/symfony#53261] [Q&A]
I recently added 

    #[Assert\EnableAutoMapping([])]

to some entities, but they're not validating, I think because the default validation group ("Default") is NotNull, even for fields that aren't persisted to the database!

Is there a way to configure this so that my entities validate?  It's particularly challenging because the ones that are failing come from a bundle trait, so I don't easily have control over its declaration.

I think I want to simply disable Default altogether, but I don't know.  I've looked at doctrine.yaml and validation.yaml and trying to pass groups into the EnableAutoMapping trait, but I'm not finding anything.

Thanks.

**Comment by MatTheCat** [comment]
The auto-mapping is Doctrine’s, so it shouldn’t consider unmapped properties. Cannot say what’s wrong without a reproducer, but it looks like `EnableAutoMapping` is not a good fit here, so you should maybe consider configuring constraints yourself.

BTW, validating your entities means your entities are allowed to be invalid, which is an anti-pattern (see https://verraes.net/2015/02/form-command-model-validation/ e.g.).
